cv.typ

```diff
@@ -2,341 +2,391 @@
-            )
-        }
+    if info.personal.profiles.len() > 0 {
+      for profile in info.personal.profiles {
+        profiles.push(box(link(profile.url, { profile.url.split("//").at(1) })))
+      }
     }
 
-    #set text(font: uservars.bodyfont, weight: "medium", size: uservars.fontsize * 1)
-    #pad(x: 0em)[
-        #profiles.join([#sym.space.en #sym.diamond.filled #sym.space.en])
-    ]
-]
+    set text(font: uservars.bodyfont, weight: "medium", size: uservars.fontsize * 1)
+    pad(x: 0em, {
+      profiles.join([#sym.space.en #sym.diamond.filled #sym.space.en])
+    })
+  },
+)
 
 #let cvheading(info, uservars) = {
-    align(center)[
-        = #info.personal.name
-        #addresstext(info, uservars)
-        #contacttext(info, uservars)
-    ]
+  align(center, {
+    [= #info.personal.name]
+
+    addresstext(info, uservars)
+    contacttext(info, uservars)
+  })
 }
 
 #let cvwork(info, isbreakable: true) = {
-    if info.work != none {block[
-        == Work Experience
-        #for w in info.work {
-            block(width: 100%, breakable: isbreakable)[
-                // line 1: company and location
-                #if w.url != none [
-                    *#link(w.url)[#w.organization]* #h(1fr) *#w.location* \
-                ] else [
-                    *#w.organization* #h(1fr) *#w.location* \
-                ]
-            ]
-            // create a block layout for each work entry
-            let index = 0
-            for p in w.positions {
-                if index != 0 {v(0.6em)}
-                block(width: 100%, breakable: isbreakable, above: 0.6em)[
-                    // parse ISO date strings into datetime objects
-                    #let start = utils.strpdate(p.startDate)
-                    #let end = utils.strpdate(p.endDate)
-                    // line 2: position and date range
-                    #text(style: "italic")[#p.position] #h(1fr)
-                    #start #sym.dash.en #end \
-                    // highlights or description
-                    #for hi in p.highlights [
-                        - #eval(hi, mode: "markup")
-                    ]
-                ]
-                index = index + 1
-            }
+  if info.at("work", default: none) != none {
+    block({
+      [== Work Experience]
+
+      for w in info.work {
+        let company = w.organization
+        let url = w.at("url", default: none)
+        let location = w.location
+
+        line_one(company, right: location, url: url)
+
+        let index = 0
+        for p in w.positions {
+          let start = p.at("startDate", default: none)
+          if start != none {
+            start = utils.str_to_date(start)
+          }
+          let end = utils.str_to_date(p.endDate)
+
+          if index != 0 { v(0.5em) }
+
+          block(above: 0.5em, {
+            line_two(p.position, right_start: start, right_end: end)
+          })
+
+          for hi in p.highlights {
+            [- #hi]
```

Comment:
Hm, this part should be reverted back to using `eval()`. This breaks the use of markup like bold or italics when present in the YAML file.

```suggestion
        for hi in p.highlights{
            [- #eval(hi, mode: "markup")]
        }
```